objects/src/accounts/delta/mod.rs

```diff
@@ -58,6 +58,15 @@ impl AccountDelta {
         Ok(Self { storage, vault, nonce })
     }
 
+    /// Merge another [AccountDelta] into this one.
+    pub fn merge(self, other: Self) -> Result<Self, AccountDeltaError> {
+        // Incoming nonce takes precedence.
+        let nonce = other.nonce.or(self.nonce);
```

Comment:
I would add a check here that incoming nonce is greater than current nonce.

Reply:
I've added the check. I'd like to add some derives to `Felt` - `PartialOrd`, `Ord` so that these comparisons are less clunky.